File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.java
import java.util.HashMap;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        String[] answer = {};
        HashMap<String, Integer> playerInfo = new HashMap<>();
        for(int i=0; i<players.length; i++){
            playerInfo.put(players[i], i);
        }
        for(int i=0; i<callings.length; i++) {
            int get = playerInfo.get(callings[i]);
            String chName = players[get-1];
            
            players[get-1] = callings[i];
            players[get] = chName;
            
            playerInfo.put(callings[i], get-1);
            playerInfo.put(chName, get);
        }
        return players;
    }
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    var score = new Map();
    players.forEach((cur, idx) => score.set(cur, idx));
    for(var i=0; i<callings.length; i++) {
        var me = callings[i];
        var myIdx = score.get(me);
        var other = players[myIdx-1];
        
        players[myIdx]  = other;
        players[myIdx-1] = me;
        
        score.set(me, score.get(me)-1);
        score.set(other, score.get(other)+1);
    }
    return players;
}